core/src/main/java/org/apache/accumulo/core/iteratorsImpl/system/HeapIterator.java

```diff
@@ -72,6 +73,10 @@ public final void next() throws IOException {
       throw new IllegalStateException("Called next() when there is no top");
     }
 
+    if (Thread.currentThread().isInterrupted()) {
```

Comment:
This could be called frequently and not sure what the implications of that are for performance.  Was thinking that when an rfile reads a new block that might be a good place to check because its naturally not called too frequently.  Found the following code.

https://github.com/apache/accumulo/blob/de479d1b314207c5c139d4ecef2bc5e43f4553a7/core/src/main/java/org/apache/accumulo/core/file/rfile/RFile.java#L886-L888


That code is already doing a check for the interrupt boolean.  Looking into this a bit further I can not find any place where an interrupt atomic boolean is set for the compaction code.  Only seeing scan code set it.  So maybe if the compaction code did set something on it iterator stack then this existing code would eventually throw an exception even when nothing is being returned by the iterator stack.

The following is the place where the scan code sets the interrupt flag.

https://github.com/apache/accumulo/blob/de479d1b314207c5c139d4ecef2bc5e43f4553a7/server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/ScanDataSource.java#L179